pkgs/development/python-modules/screeninfo/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, buildPythonApplication, fetchPypi, isPy36, dataclasses, libX11, libXinerama, libXrandr }:
+
+buildPythonApplication rec {
+  pname = "screeninfo";
+  version = "0.6.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0vcw54crdgmbzwlrfg80kd1a8p9i10yks8k0szzi0k5q80zhp8xz";
+  };
+
+  # dataclasses is a compatibility shim for python 3.6 ONLY
+  patchPhase = if isPy36 then "" else ''
+    substituteInPlace setup.py \
+      --replace "\"dataclasses\"," ""
```

Comment:
It's already conditional on `isPy36` - it's not necessary for Python 3.7 (and newer) because `dataclasses` was merged into core. If you look at the `dataclasses` python module, you'll see that it only builds if `isPy36` is true.